nesta_ds_utils/loading_saving/S3.py

```diff
@@ -294,64 +365,85 @@ def download_obj(
         bucket (str): Bucket's name.
         path_from (str): Path to data in S3.
         download_as (str, optional): Type of object downloading. Choose between
-        ('dataframe', 'dict', 'list', 'str', 'np.array'). Not needed for 'pkl files'.
+        ('dataframe', 'geodataframe', 'dict', 'list', 'str', 'np.array'). Not needed for 'pkl files'.
         kwargs_boto (dict, optional): Dictionary of kwargs for boto3 function 'download_fileobj'.
         kwargs_reading (dict, optional): Dictionary of kwargs for reading data.
 
     Returns:
-        any: Donwloaded data.
+        any: Downloaded data.
     """
-    if not path_from.endswith(tuple([".csv", ".parquet", ".json", ".txt", ".pkl"])):
-        raise Exception(
+    if not path_from.endswith(
+        tuple(
+            [".csv", ".parquet", ".json", ".txt", ".pkl", ".geojson", ".xlsx", ".xlsm"]
+        )
+    ):
+        raise NotImplementedError(
             "This file type is not currently supported for download in memory."
         )
     s3 = boto3.client("s3")
     fileobj = io.BytesIO()
     s3.download_fileobj(bucket, path_from, fileobj, **kwargs_boto)
     fileobj.seek(0)
     if download_as == "dataframe":
-        if path_from.endswith(tuple([".csv", ".parquet"])):
+        if path_from.endswith(tuple([".csv", ".parquet", ".xlsx", ".xlsm"])):
             return _fileobj_to_df(fileobj, path_from, **kwargs_reading)
         else:
-            raise Exception(
+            raise NotImplementedError(
                 "Download as dataframe currently supported only "
-                "for 'csv' and 'parquet'."
+                "for 'csv','parquet','xlsx' and 'xlsm'."
+            )
+    elif download_as == "geodataframe":
```

Comment:
```suggestion
    elif download_as.lower() in ["geodataframe", "gdf", "geodf", "geo_df"]:
```
geodataframe may be lengthly for end users

Reply:
I think its probably simpler to just have one option? We could go with geodf?